uktrade/data-hub-frontend
# Add more complex fields to opportunity details form

## Description of change
This PR introduces the more complex fields on the Opportunities Details form. It has been created following [this guideline](https://x1jxy1.axshare.com/#g=1&p=ukio_oppdetails_edit), however, a couple of things had to be changed due to elements in the wireframes not being available. These have been replaced with appropriate alternatives.

## Test instructions

Go to the Investments tab in the main nav, click on UK Opportunities, click on an opportunity. You should see two toggles, click on the Details toggle. You should see a summary table and an edit button. When clicking the edit button, you should see the form. You should be able to complete the form, with only the name being mandatory. All fields that have existing data should have their existing data as initial values in all of the fields. You should be able to submit the form and be redirected back to the summary table that shows your new values.

![Screenshot 2021-07-28 at 15 02 20](https://user-images.githubusercontent.com/46787754/127336266-b10b4030-a682-4d09-b7b1-d6767fa96c9b.png)
![Screenshot 2021-07-28 at 15 02 32](https://user-images.githubusercontent.com/46787754/127336273-f25dc0af-13c3-41fe-823a-e4cd11e3bbb9.png)
![Screenshot 2021-07-28 at 15 02 45](https://user-images.githubusercontent.com/46787754/127336283-0e1117f7-2c44-4e70-b618-587738eca239.png)
![Screenshot 2021-07-28 at 15 03 09](https://user-images.githubusercontent.com/46787754/127336286-e4a96e89-afe5-47a5-a819-36e10f35087f.png)


[//]: # "When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/master/docs/Code%20review%20guidelines.md"
